afdko/Tools/SharedData/FDKScripts/ufoTools.py

```diff
@@ -364,2003 +366,2231 @@ def debugMsg(*args):
+            except KeyError:
+                raise UFOParseError(
+                    "'%s' component glyph is missing from "
+                    "contents.plist." % compGlyphName)
+        else:
+            # If we are not necessarily using the default layer for the main
+            # glyph, then a missing component may not have been processed, and
+            # may just be in the default layer. We need to look for component
+            # glyphs in the src list first, then in the defualt layer.
+            try:
+                compGlyphFilePath = self.getGlyphSrcPath(compGlyphName)
+                if not os.path.exists(compGlyphFilePath):
+                    compGlyphFilePath = self.getGlyphDefaultPath(compGlyphName)
+            except KeyError:
+                try:
+                    compGlyphFilePath = self.getGlyphDefaultPath(compGlyphName)
+                except KeyError:
+                    raise UFOParseError(
+                        "'%s' component glyph is missing from "
+                        "contents.plist." % compGlyphName)
+
+        if not os.path.exists(compGlyphFilePath):
+            raise UFOParseError(
+                "'%s' component file is missing: '%s'." %
+                (compGlyphName, compGlyphFilePath))
+
+        etRoot = ET.ElementTree()
+        glifXML = etRoot.parse(compGlyphFilePath)
+        outlineXML = glifXML.find("outline")
+        return outlineXML
+
+    def close(self):
+        if self.hashMapChanged:
+            self.writeHashMap()
+            self.hashMapChanged = 0
+        return
+
+    def clearHashMap(self):
+        self.hashMap = {kAdobHashMapVersionName: kAdobHashMapVersion}
+        hashDir = os.path.join(self.parentPath, "data")
+        if not os.path.exists(hashDir):
+            return
+
+        hashPath = os.path.join(hashDir, kAdobHashMapName)
+        if os.path.exists(hashPath):
+            os.remove(hashPath)
+
+    def setWriteToDefault(self):
+        self.useProcessedLayer = False
+        self.writeToDefaultLayer = True
+        self.glyphWriteDir = self.glyphDefaultDir
 
 
+def parseGlyphOrder(filePath):
+    orderMap = None
+    if os.path.exists(filePath):
+        publicOrderDict, temp = parsePList(filePath, kPublicGlyphOrderKey)
+        if publicOrderDict:
+            orderMap = {}
+            glyphList = publicOrderDict[kPublicGlyphOrderKey]
+            numGlyphs = len(glyphList)
+            for i in range(numGlyphs):
+                orderMap[glyphList[i]] = i
+    return orderMap
+
+
+def parsePList(filePath, dictKey=None):
+    # If dictKey is defined, parse and return only the data for that key.
+    # This helps avoid needing to parse all plist types. We need to support
+    # only data for known keys amd lists.
+    plistDict = {}
+    plistKeys = []
+
+    # We uses this rather than the plistlib in order to get a list that
+    # allows preserving key order.
+    fp = open(filePath, "r")
+    data = fp.read()
```

Comment:
I think we should remove *all* references to `lastPathOp`.
I count 16 assignments and zero actual usage.